### PR TITLE
Add `WritableTopicStore`

### LINKED ISSUE #4923: Implement `WritableTopicStore`

### PR DESCRIPTION
**Pull request by Neeharika-Sompalli**
- Adds a `WritableTopicSore` that can put topic into state. (Did not add remove/update methods and will be added later)
- Added `Topic` , `TopicBuilder` entities

Fixes #4923

### CODE REVIEW

File: hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/entity/TopicBuilderImpl.java [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+    private HederaKey adminKey;
+    private HederaKey submitKey;
+    private String memo;
+    private long autoRenewAccountNumber;
+    private long autoRenewSecs;
+    private long expiry;
+    private boolean deleted;
+    private long sequenceNumber;
+    /**
+     * Create a builder for creating {@link Account}s, using the given copy as the basis for all
```
Comment (tinker-michaelj) [inline hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/entity/TopicBuilderImpl.java:44]: Copy/paste (`Account`).

File: hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/entity/TopicImpl.java [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+        String memo,
+        long autoRenewAccountNumber,
+        long autoRenewSecs,
+        long expiry,
+        boolean deleted,
+        long sequenceNumber)
+        implements Topic {
+
+    @Override
+    public long shardNumber() {
```
Comment (tinker-michaelj) [inline hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/entity/TopicImpl.java:41]: I'm pretty sure we'll never need to store shard or realm at the entity level; I think those are properties of the ledger as a whole.

File: hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/entity/TopicBuilderImpl.java [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+import com.hedera.node.app.spi.key.HederaKey;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+
+/**
+ * An implementation of {@link TopicBuilder} for building Topic instances. This class is
+ * <strong>not</strong> exported from the module.
+ * module
+ */
+public class TopicBuilderImpl implements TopicBuilder {
```
Comment (hendrikebbers) [inline hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/entity/TopicBuilderImpl.java:33]: We should discuss if we want to move builders & record (plain data objects) directly in the API layer.

Comment (Neeharika-Sompalli) [reply hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/entity/TopicBuilderImpl.java:32]: Yes, in my next PR I will move `Account` And `AccountBuilder` to SPI. I am not sure if all entities need to be moved though. Agreed, I will bring this up in next meeting. Thanks for pointing out @hendrikebbers

**Review by tinker-michaelj — DISMISSED** [review]
LGTM!

**Review by mhess-swl — APPROVED** [review]

**Review by kimbor — APPROVED** [review]

**Review by mhess-swl — APPROVED** [review]

**Review by hendrikebbers — APPROVED** [review]

**Pull request merged**